src/ItemList.js

```diff
@@ -60,12 +127,32 @@ function ItemList(props) {
             </button>
           )}
           <h2>Shopping List:</h2>
+
           <form>
-            <ul>
-              {props.list.map((item) => (
-                <Item key={item.id} userToken={props.userToken} item={item} />
-              ))}
-            </ul>
+            {Object.entries(queryObj).map(([key, value]) => {
+              return (
+                value.length > 0 && (
+                  <>
+                    <h3>
+                      {key === 'week' && 'Items to buy in the next week:'}
+                      {key === 'month' && 'Items to buy in the next month:'}
+                      {key === 'longer' &&
+                        'Items to buy in the distant future:'}
+                      {key === 'inactive' && "Purchases we can't predict yet:"}
+                    </h3>
+                    <ul>
+                      {value.map((item) => (
+                        <Item
+                          key={item.id}
+                          userToken={props.userToken}
+                          item={item}
+                        />
+                      ))}
+                    </ul>
```

Comment:
Love it!

Reply:
😍 Looks so clean, well done cleaning this up, Daria!